internal/daemon/controller/handlers/workers/worker_service_test.go

```diff
@@ -747,3 +753,404 @@ func TestUpdate_BadVersion(t *testing.T) {
+			req: &pbs.CreateWorkerRequest{
+				Item: &workers.Worker{
+					ScopeId:              scope.Global.String(),
+					NodeCredentialsToken: &wrapperspb.StringValue{Value: fetchReqFn()},
+				},
+			},
+			wantErr:         true,
+			wantErrContains: "non-existent scope \"splat-auth-scope\"",
+		},
+		{
+			name:    "invalid-base58-encoding",
+			service: testSrv,
+			scopeId: scope.Global.String(),
+			req: &pbs.CreateWorkerRequest{
+				Item: &workers.Worker{
+					ScopeId:              scope.Global.String(),
+					NodeCredentialsToken: &wrapperspb.StringValue{Value: "invalid;semicolon"},
+				},
+			},
+			wantErr:         true,
+			wantErrContains: "error decoding node_credentials_token",
+		},
+		{
+			name:    "invalid-marshal",
+			service: testSrv,
+			scopeId: scope.Global.String(),
+			req: &pbs.CreateWorkerRequest{
+				Item: &workers.Worker{
+					ScopeId:              scope.Global.String(),
+					NodeCredentialsToken: &wrapperspb.StringValue{Value: "notNodeCreds"},
+				},
+			},
+			wantErr:         true,
+			wantErrContains: "error unmarshaling node_credentials_token",
+		},
+		{
+			name: "create-error",
+			service: func() Service {
+				repoFn := func() (*servers.Repository, error) {
+					return servers.NewRepository(rw, &db.Db{}, testKms)
+				}
+				testSrv, err := NewService(testCtx, repoFn, iamRepoFn)
+				require.NoError(t, err, "Error when getting new worker service.")
+				return testSrv
+			}(),
+			scopeId: scope.Global.String(),
+			req: &pbs.CreateWorkerRequest{
+				Item: &workers.Worker{
+					ScopeId:              scope.Global.String(),
+					Name:                 &wrapperspb.StringValue{Value: "success"},
+					Description:          &wrapperspb.StringValue{Value: "success-description"},
+					NodeCredentialsToken: &wrapperspb.StringValue{Value: fetchReqFn()},
+				},
+			},
+			wantErr:         true,
+			wantErrContains: "error creating worker",
+		},
+		{
+			name: "bad-repo-function-in-create",
+			service: func() Service {
+				cnt := 0
+				repoFn := func() (*servers.Repository, error) {
+					cnt = cnt + 1
+					switch {
+					case cnt > 1:
+						return nil, errors.New(testCtx, errors.Internal, "bad-repo-function", "error creating repo")
+					default:
+						return servers.NewRepository(rw, rw, testKms)
+					}
+				}
+				testSrv, err := NewService(testCtx, repoFn, iamRepoFn)
+				require.NoError(t, err, "Error when getting new worker service.")
+				return testSrv
+			}(),
+			scopeId: scope.Global.String(),
+			req: &pbs.CreateWorkerRequest{
+				Item: &workers.Worker{
+					ScopeId:              scope.Global.String(),
+					Name:                 &wrapperspb.StringValue{Value: "success"},
+					Description:          &wrapperspb.StringValue{Value: "success-description"},
+					NodeCredentialsToken: &wrapperspb.StringValue{Value: fetchReqFn()},
+				},
+			},
+			wantErr:         true,
+			wantErrContains: "error creating worker: bad-repo-function: error creating repo",
+		},
+		{
```

Comment:
I appreciate the suggestion and I've seen this pattern in a few unit tests since I've come back to boundary... perhaps the pattern will find a home with me, but so far I'm not quite ready to adopt.